01/ht/edithpifpaf/01/01.js

```diff
@@ -0,0 +1,31 @@
+function calc(x) {
+  return function calculate(z) {
+    if (z==='+') {
+      return function sum(y) {
+        return checkIfNumber(x) + checkIfNumber(y);
+      }
+    } else if (z==='-') {
+      return function subtract(y) {
+        return checkIfNumber(x) - checkIfNumber(y);
+      }
+    } else if (z==='*') {
+      return function multiply(y) {
+        return checkIfNumber(x) * checkIfNumber(y);
+      }
+    } else if (z==='/') {
+      return function divide(y) {
+        return checkIfNumber(x) / checkIfNumber(y);
+      }
+   } else {
+      alert (' Not an operator! ');
+    }
+  }
+}
+
+function checkIfNumber(number) {
```

Comment:
the same about naming, but we'll discuss it on Saturday

Waiting other solutions )